tests/map/test-map.c

```diff
@@ -0,0 +1,111 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "map.h"
+
+void swap(int *x, int *y)
+{
+    int tmp = *x;
+    *x = *y;
+    *y = tmp;
+}
+
+enum { N_NODES = 10000 };
+
+bool test_mix_insert()
+{
+    bool failed = false;
+
+    map_t tree = map_init(int, int, map_cmp_uint);
+
+    int key[N_NODES], val[N_NODES];
+
+    /*
+     *  Generate data for insertion
+     */
+    for (int i = 0; i < N_NODES; i++) {
+        key[i] = i;
+        val[i] = i + 1;
+    }
+
+    /* TODO: This is not a reconmended way to randomize stuff, just a simple
+     * test. Using MT19937 might be better
+     */
+    srand((unsigned) time(NULL));
+    for (int i = 0; i < N_NODES; i++) {
+        int pos_a = rand() % N_NODES;
+        int pos_b = rand() % N_NODES;
+        swap(&key[pos_a], &key[pos_b]);
+        swap(&val[pos_a], &val[pos_b]);
+    }
+
+    /* add first 1/2 items */
+    for (int i = 0; i < N_NODES / 2; i++) {
+        map_iter_t my_it;
+        map_insert(tree, key + i, val + i);
+        map_find(tree, &my_it, key + i);
+        if (NULL == my_it.node) {
+            failed = true;
+        }
+        assert((*(int *) (my_it.node->data)) == val[i]);
+    }
+
+    /* remove first 1/4 items */
+    for (int i = 0; i < N_NODES / 4; i++) {
+        map_iter_t my_it;
+        map_find(tree, &my_it, key + i);
+        if (!map_at_end(tree, &my_it)) {
+            map_erase(tree, &my_it);
+            map_find(tree, &my_it, key + i);
+            if (NULL != my_it.node) {
+                assert(1);
+                failed = true;
+            }
+        }
+    }
+
+    /* add the rest */
+    for (int i = N_NODES / 2 + 1; i < N_NODES; i++) {
+        map_iter_t my_it;
+        map_insert(tree, key + i, val + i);
+        map_find(tree, &my_it, key + i);
+        if (NULL == my_it.node) {
+            failed = true;
+        }
+        assert((*(int *) (my_it.node->data)) == val[i]);
+    }
+
+
+    /* remove 2nd quarter of items */
+    for (int i = N_NODES / 4 + 1; i < N_NODES / 2; i++) {
+        map_iter_t my_it;
+        map_find(tree, &my_it, key + i);
+        if (!map_at_end(tree, &my_it)) {
+            map_erase(tree, &my_it);
+            map_find(tree, &my_it, key + i);
+            if (NULL != my_it.node) {
+                assert(1);
+                failed = true;
+            }
+        }
+    }
+
+    map_clear(tree);
+    map_delete(tree);
+
+    return !failed;
+}
+
+int main(int argc, char *argv[])
+{
+    (void) argc;
+    (void) argv;
+    bool passed = test_mix_insert();
+    printf("%d", passed);
```

Comment:
If not passed, return non-zero value.

Reply:
What is the purpose of this?

Pass --> 1 
!Pass --> 999

Reply:
No, `0` on success; `1` (or other non-zero values) on failure. It is a convention of UNIX.